hibernate-core/src/main/java/org/hibernate/type/format/jakartajson/JsonBJsonFormatMapper.java

```diff
@@ -33,6 +34,10 @@ public JsonBJsonFormatMapper(Jsonb jsonb) {
 
 	@Override
 	public <T> T fromString(CharSequence charSequence, JavaType<T> javaType, WrapperOptions wrapperOptions) {
+		if ( charSequence == null || "null".contentEquals(charSequence) ) {
+			// take H2 for example "null format json" will produce literal "null" instead of database NULL
```

Comment:
Instead of special casing the `null` string for this H2 "quirk" I would suggest to change `H2FormatJsonJdbcType` to use `cast(... as json)` instead of the `... format json` syntax. I don't know why we chose that particular syntax, maybe because it is documented.

At least in my quick tests, using the cast syntax worked fine and produces a SQL `null` if `null` is passed.

Reply:
`'[]' format json` will produce expected `[]`, but `cast('[]' as json)` will produce unwanted `"[]"`.